castai/resource_castai_eks_cluster.go

```diff
@@ -261,8 +266,16 @@ func resourceCastaiEKSClusterDelete(ctx context.Context, data *schema.ResourceDa
 
 		if clusterResponse.JSON200.CredentialsId != nil && agentStatus != sdk.ClusterAgentStatusDisconnected {
 			log.Printf("[INFO] Disconnecting cluster.")
+			deleteNodes := false
```

Comment:
nitpick: all this "get optional bool or default value" logic breaks the flow here, wrapping into helper func would help a lot:

```go
response, err := client.ExternalClusterAPIDisconnectClusterWithResponse(ctx, clusterId, sdk.ExternalClusterAPIDisconnectClusterJSONRequestBody{
	DeleteProvisionedNodes: getOptionalBool(data, FieldEKSClusterDeleteNodesOnDisconnect, false)
})
//...

func getOptionalBool(data, field, defaultValue) {
}
```

Reply:
good point, fixed